tests/e2e/common/parameterized_query_tests.py

```diff
@@ -0,0 +1,181 @@
+from datetime import datetime
+from decimal import Decimal
+
+import pytz
+
+from databricks.sql.client import Connection
+from databricks.sql.utils import DbSqlParameter, DbSqlType
+
+
+class PySQLParameterizedQueryTestSuiteMixin:
+    """Namespace for tests of server-side parameterized queries"""
+
+    def test_parameterized_query_named_and_inferred_e2e(self):
```

Comment:
A minor point of testing style that you can choose to ignore (as here is not really the worst violation I've seen on this point): unless the setup is very expensive (in which case consider a shared setup), a test should validate one thing, so that if the test fails, you know immediately where it failed (if not why).  Here we are bundling all of the column types together; this is fine if we're pretty confident that the asserts will either all fail or all pass.  If this is not the case, might be better to test each type individually.

Reply:
Agree that will make this easier to evaluate and extend in the future. Will ping you for a re-review once it's done.

Reply:
Done in a8841f884b8576561961cf69ec356ecf15ca6752